src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -589,6 +611,11 @@ impl ConstCx<'_, 'tcx> {
     }
 }
 
+/// Checks MIR for const-correctness, using `ConstCx`
+/// for value qualifications, and accumulates writes of
+/// rvalue/call results to locals, in `local_qualif`.
+/// For functions (constant or not), it also records
+/// candidates for promotion in `promotion_candidates`.
```

Comment:
Not just functions though, I think for everything? Promotion still has to do something in `const`/`static`, it's just slightly different than at runtime.

Reply:
No idea, I just moved that comment up, I did not write it.^^

Reply:
There's a comment in the file though saying
```
            // No need to do anything in constants and statics, as everything is "constant" anyway
            // so promotion would be useless.
            if self.mode != Mode::Static && self.mode != Mode::Const {
```
So it does seem to not do promotion the same way for statics and consts?